### PR TITLE
ENH: Improve module access scripts in slicer.util

### PR DESCRIPTION
**Pull request by lassoan**
Add function to return module logic.
Improve module widget function to return the actual Python class instead of the C++ wrapper.
Use exceptions for error handling (instead of just printing error message) to make sure that the error is noticed and the most relevant error message is displayed.
Add missing function documentation.

### CODE REVIEW

**Review by pieper — APPROVED** [review]
Is there any compatibility issue related to returning the python class compared to the c++ wrapper?

**Comment by lassoan** [comment]
Yes, the behavior is now slightly different, but probably very few people used these functions, as they were not advertised anywhere, not documented, and they did not do that much (they just replaced a single-line operation with a single command). Now that it can do a bit more, it makes more sense to recommend it over the basic C++ API in examples and tutorials.

File: Base/Python/slicer/util.py [hunk 1]
```diff
@@ -194,8 +194,7 @@ def lookupTopLevelWidget(objectName, verbose = True):
     if hasattr(w,'objectName'):
       if w.objectName == objectName: return w
   if verbose:
-    import sys
-    print("Failed to obtain reference to '%s'" % objectName, file=sys.stderr)
+    raise RuntimeError("Failed to obtain reference to '%s'" % objectName)
```
Comment (jcfr) [inline Base/Python/slicer/util.py:197]: Now we raise an exception. It should probably be done all the time and not only when verbose is True.

File: Base/Python/slicer/util.py [hunk 2]
```diff
@@ -205,7 +204,7 @@ def pythonShell(verbose = True):
   from slicer import app
   console = app.pythonConsole()
   if not console and verbose:
-    print("Failed to obtain reference to python shell", file=sys.stderr)
+    raise RuntimeError("Failed to obtain reference to python shell")
```
Comment (jcfr) [inline Base/Python/slicer/util.py:207]: Same comment here.

File: Base/Python/slicer/util.py [hunk 3]
```diff
@@ -928,64 +927,131 @@ def saveScene(filename, properties={}):
 #
 
 def moduleSelector():
+  """Return module selector widget.
+  Throws a RuntimeError exception if there is no module selector (for example, the application runs without a main window).
```
Comment (jcfr) [inline Base/Python/slicer/util.py:931]: There is a way to document exception. See https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format

File: Base/Python/slicer/util.py [hunk 4]
```diff
@@ -928,64 +927,131 @@ def saveScene(filename, properties={}):
...
   module = app.moduleManager().module(moduleName)
   if not module:
-    import sys
-    print("Could not find module with name '%s" % moduleName, file=sys.stderr)
-    return None
+    raise RuntimeError("Could not find module with name '%s'" % moduleName)
   return module
 
 def getModuleGui(module):
+  """Deprecated. Use getModuleWidget function instead"""
```
Comment (jcfr) [inline Base/Python/slicer/util.py:981]: I suggest to use the 'deprecated directive.  See https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated

File: Base/Python/slicer/util.py [hunk 5]
```diff
@@ -928,64 +927,131 @@ def saveScene(filename, properties={}):
...
+      # This widget does not have a logic instance in its widget class
+      logic = None
+  else:
+    logic = module.logic()
+  if not logic:
+    raise RuntimeError("Could not find module widget representation with name '%s'" % module.name)
+  return logic
 
 def modulePath(moduleName):
+  """Get module logic object.
```
Comment (jcfr) [inline Base/Python/slicer/util.py:1049]: This docstring seems incorrect

**Pull request merged**